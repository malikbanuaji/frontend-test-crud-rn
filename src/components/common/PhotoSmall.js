import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import {Mixins, Typography} from '../../styles';

export default function PhotoSmall({photo, firstName, lastName, style}) {
  return (
    <View style={[style]}>
      {photo === 'N/A' ? (
        <View style={[styles.emptyPhoto]}>
          <Text style={styles.emptyText}>
            {`${firstName.substring(0, 1)}${lastName.substring(
              0,
              1,
            )}`.toUpperCase()}
          </Text>
        </View>
      ) : (
        <Image
          style={styles.photo}
          source={{uri: photo}}
          width={40}
          height={40}
        />
      )}
    </View>
  );
}

PhotoSmall.defaultProps = {
  firstName: '',
  lastName: '',
  photo: 'N/A',
};

const styles = StyleSheet.create({
  photo: {
    ...Mixins.photoSmallDimension,
    ...Mixins.photo,
  },
  emptyPhoto: {
    ...Mixins.emptyPhoto,
    ...Mixins.photoSmallDimension,
  },
  emptyText: {
    ...Typography.photoSmallEmptyText,
  },
});
