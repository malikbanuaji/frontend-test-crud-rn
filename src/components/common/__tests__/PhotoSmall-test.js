import React from 'react';
import renderer from 'react-test-renderer';
import PhotoSmall from '../PhotoSmall';

test('renders correctly with image', () => {
  const tree = renderer
    .create(
      <PhotoSmall
        photo={
          'https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png'
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without image', () => {
  const tree = renderer.create(<PhotoSmall photo={'N/A'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without firstName', () => {
  const tree = renderer.create(<PhotoSmall lastName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly without lastName', () => {
  const tree = renderer.create(<PhotoSmall firstName={'Doe'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with full props', () => {
  const tree = renderer
    .create(
      <PhotoSmall
        firstName={'Doe'}
        lastName={'Doe'}
        photo={
          'https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png'
        }
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
