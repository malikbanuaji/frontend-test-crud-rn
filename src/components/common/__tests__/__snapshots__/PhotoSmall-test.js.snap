// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with full props 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <Image
    height={40}
    source={
      Object {
        "uri": "https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png",
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": 40,
        "overflow": "hidden",
        "width": 40,
      }
    }
    width={40}
  />
</View>
`;

exports[`renders correctly with image 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <Image
    height={40}
    source={
      Object {
        "uri": "https://mattermost.com/wp-content/uploads/2018/10/React_Native_Logo.png",
      }
    }
    style={
      Object {
        "borderRadius": 8,
        "height": 40,
        "overflow": "hidden",
        "width": 40,
      }
    }
    width={40}
  />
</View>
`;

exports[`renders correctly without firstName 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#93b5e1",
          "borderRadius": 8,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
        }
      }
    >
      D
    </Text>
  </View>
</View>
`;

exports[`renders correctly without image 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#93b5e1",
          "borderRadius": 8,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders correctly without lastName 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#93b5e1",
          "borderRadius": 8,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
        }
      }
    >
      D
    </Text>
  </View>
</View>
`;
